src/io/hdf/hdf_path.cpp

```diff
@@ -23,8 +23,18 @@ hdf_path& hdf_path::append(const hdf_path& p)
 {
   if (p.has_root_directory())
     path_.clear();
-  else if (path_.back() != '/')
-    path_.push_back('/');
+  else
+  {
+    if (path_.size() > 0)
+    {
+      if (path_.back() != '/')
+        path_.push_back('/');
+    }
+    else
+    {
+      path_.push_back('/');
```

Comment:
This line adds a new execution behavior. Do you have an example explaining that this is line is needed?

Reply:
Will fix.